src/rdkafka_transport.c

```diff
@@ -475,13 +476,16 @@ void rd_kafka_transport_ssl_term (void) {
 
 	CRYPTO_set_id_callback(NULL);
 	CRYPTO_set_locking_callback(NULL);
-        CRYPTO_cleanup_all_ex_data();
+    EVP_cleanup();
+    CRYPTO_cleanup_all_ex_data();
+    ERR_free_strings();
```

Comment:
remove this call, it is not safe to do when SSL is used by multiple libs in the same process, see #815

Reply:
I am confused here?  Do you want me to remove all of the changes here to this function?  If not can you explain what is not safe here?

Reply:
@edenhill  can you please comment on this so I can address this too.

Reply:
What @edenhill means, @noahdav, is that this may cause conflicts with other programs or libraries that are also using OpenSSL; #815 or https://www.postgresql.org/message-id/20081203200427.0A23B7545A4@cvs.postgresql.org are examples of this, although I am not sure how `ERR_free_strings` specifically is affected.

Reply:
Thanks @dzuelke for the clarification.  The article that you liked talks about having callbacks set to unloaded modules.  In this case I do not see how this will happen.  I am not understanding how ERR_free_strings is different from CRYPTO_cleanup_all_ex_data or the other cleanup routines here.  If you feel strongly I can remove this but I am not clear to the issue by calling ERR_free_strings.